components/raftstore/src/store/peer.rs

```diff
@@ -1574,7 +1574,12 @@ where
 
         if !self.raft_group.has_ready() {
             // Generating snapshot task won't set ready for raft group.
-            if let Some(gen_task) = self.mut_store().take_gen_snap_task() {
+            if let Some(mut gen_task) = self.mut_store().take_gen_snap_task() {
+                if let Some(progress) = self.raft_group.status().progress {
+                    if progress.iter().any(|(_, pr)| pr.matched == 0) {
```

Comment:
not familiar with the code here. what does `pr.matched == 0` mean?

Reply:
It relies on the fact that when a peer is uninitialized (e.g. created by load balance), the last index of the peer is 0 which makes the matched index to be 0.

Reply:
PTAL @hicqu

Reply:
Add comment for it?